client/state/selectors/get-current-user-payment-methods.js

```diff
@@ -10,7 +10,7 @@ import { lowerCase, upperCase } from 'lodash';
  * Internal dependencies
  */
 import { getCurrentUserLocale } from 'state/current-user/selectors';
-import { getGeoCountryShort } from 'state/geo/selectors';
+import { requestGeoLocation } from 'state/data-layer/http-data/getters';
```

Comment:
Could we draw a more visible line between framework and application code? Inside the `state/data-layer/http-data` module, there is a mixture of both.

Reply:
I was struggling to know where to put these. In some cases there will be no getter. In other cases we'll have a single function and I don't think that warrants a separate directory like `geo` had before this patch.

Any ideas on organizing these functions?

Reply:
Then maybe `state/data-getters/index.js` could be a good home for these tiny getters? My request was to separate more clearly between framework modules and app modules that use them. Whether it's `state/geo`, `state/data-getters` or `/lib/...`, I don't really have a strong preference. What's your view about the "framework vs app" separation, do you think it makes sense to try to achieve it here?

Reply:
right now these are not like the data layer handlers. they are tiny utility helper functions and will be short-lived and rarely-called and the only reason I made it is because `geo` gets requested twice. I think `state/data-getters` can make sense. personally I don't see much conflict with the framework/app distinction because these two things (the subsystem and the helpers) are strongly coupled together; on the same note I don't care too much about changing how it's done (I just don't see reason to made a directory structure for every five to ten line function).